internal/services/appservice/helpers/function_app_schema.go

```diff
@@ -1984,6 +2020,12 @@ func FlattenSiteConfigLinuxFunctionApp(functionAppSiteConfig *web.SiteConfig) (*
 		VnetRouteAllEnabled:     utils.NormaliseNilableBool(functionAppSiteConfig.VnetRouteAllEnabled),
 	}
 
+	publicNetworkAccess := true
+	if functionAppSiteConfig.PublicNetworkAccess != nil && *functionAppSiteConfig.PublicNetworkAccess == "Disabled" {
+		publicNetworkAccess = false
+	}
+	result.PublicNetworkAccessEnabled = publicNetworkAccess
```

Comment:
I think this can be simplified to 
```suggestion
	result.PublicNetworkAccessEnabled = strings.EqualsFold(pointer.From(functionAppSiteConfig.PublicNetworkAccess), "enabled")
```